internal/controller/reinstall/helper.go

```diff
@@ -0,0 +1,228 @@
+						ci.WaveAnnotation: fmt.Sprintf("%d", manifest.SyncWave),
+					},
+				},
+			})
+			if err != nil {
+				return nil, fmt.Errorf("encountered an error while converting ManagedCluster to RenderedObject, err: %w",
+					err)
+			}
+			return obj, nil
+		}
+	}
+	return nil, nil
+}
+
+func findReinstallStatusCondition(
+	clusterInstance *v1alpha1.ClusterInstance,
+	conditionType v1alpha1.ClusterInstanceConditionType,
+) *metav1.Condition {
+	if clusterInstance.Status.Reinstall == nil {
+		return nil
+	}
+
+	return meta.FindStatusCondition(clusterInstance.Status.Reinstall.Conditions, string(conditionType))
+}
+
+func setReinstallStatusCondition(clusterInstance *v1alpha1.ClusterInstance, condition metav1.Condition) bool {
+	if clusterInstance.Status.Reinstall == nil {
+		return false
+	}
+
+	return meta.SetStatusCondition(&clusterInstance.Status.Reinstall.Conditions, condition)
+}
+
+func conditionSetter(
+	conditionType v1alpha1.ClusterInstanceConditionType,
+	conditionStatus metav1.ConditionStatus,
+	conditionReason v1alpha1.ClusterInstanceConditionReason,
+	message string,
+) *metav1.Condition {
+	return &metav1.Condition{
+		Type:    string(conditionType),
+		Status:  conditionStatus,
+		Reason:  string(conditionReason),
+		Message: message,
+	}
+}
+
+func reinstallRequestProcessedConditionStatus(
+	status metav1.ConditionStatus,
+	reason v1alpha1.ClusterInstanceConditionReason,
+	message string,
+) *metav1.Condition {
+	return conditionSetter(v1alpha1.ReinstallRequestProcessed, status, reason, message)
+}
+
+func reinstallRequestValidatedConditionStatus(
+	status metav1.ConditionStatus,
+	reason v1alpha1.ClusterInstanceConditionReason,
+	message string,
+) *metav1.Condition {
+	return conditionSetter(v1alpha1.ReinstallRequestValidated, status, reason, message)
+}
+
+func reinstallPreservationDataBackedupConditionStatus(
+	status metav1.ConditionStatus,
+	reason v1alpha1.ClusterInstanceConditionReason,
+	message string,
+) *metav1.Condition {
+	return conditionSetter(v1alpha1.ReinstallPreservationDataBackedup, status, reason, message)
+}
+
+func reinstallClusterIdentityDataDetectedConditionStatus(
+	status metav1.ConditionStatus,
+	reason v1alpha1.ClusterInstanceConditionReason,
+	message string,
+) *metav1.Condition {
+	return conditionSetter(v1alpha1.ReinstallClusterIdentityDataDetected, status, reason, message)
+}
+
+func reinstallRenderedManifestsDeletedConditionStatus(
+	status metav1.ConditionStatus,
+	reason v1alpha1.ClusterInstanceConditionReason,
+	message string,
+) *metav1.Condition {
+	return conditionSetter(v1alpha1.ReinstallRenderedManifestsDeleted, status, reason, message)
+}
+
+func reinstallPreservationDataRestoredConditionStatus(
+	status metav1.ConditionStatus,
+	reason v1alpha1.ClusterInstanceConditionReason,
+	message string,
+) *metav1.Condition {
+	return conditionSetter(v1alpha1.ReinstallPreservationDataRestored, status, reason, message)
+}
+
+func patchClusterInstanceReinstallStatus(
```

Comment:
In almost all the places you call this you're also logging.

Seems a bit unnecessary to have this extra function here.

Reply:
Fair enough - removed and added logging where missing.